src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1338,9 +1464,13 @@ static void processContextRegistrationElement (BSONObj cr, EntityIdVector enV, A
 
     /* Note that attributes can be included only if at least one entity has been found */
     if (crr.contextRegistration.entityIdVector.size() > 0) {
-        std::vector<BSONElement> queryAttrV = cr.getField(REG_ATTRS).Array();
-        for (unsigned int ix = 0; ix < queryAttrV.size(); ++ix) {
+        if (cr.hasField(REG_ATTRS)) /* To prevent registration in the E-<null> style */
+        {
+          std::vector<BSONElement> queryAttrV = cr.getField(REG_ATTRS).Array();
+          for (unsigned int ix = 0; ix < queryAttrV.size(); ++ix)
+          {
             processAttribute(&crr, attrL, queryAttrV[ix].embeddedObject());
+          }
```

Comment:
The change is the `hasField()` checking. With the new logic (I'm not sure why... I did when I started working on this and I dont' remember very well... something related with E-null> I have some problems if I don't include that.